pallets/staking-rewards/src/tests.rs

```diff
@@ -1183,6 +1183,57 @@ fn unclaimed_rewards_burn() {
     });
 }
 
+#[test]
+fn empty_rewards_pool_causes_inflation() {
+    let (target_inflation, ideal_stake, _, non_stakeable) = sensible_defaults();
+    let pool_balance = 0; // empty rewards pool
+    let mut ext = with_parameters(target_inflation, ideal_stake, pool_balance, non_stakeable);
+    ext.execute_with(|| {
+        // Getting up-to-date data on era duration (they may differ from runtime constants)
+        let sessions_per_era = <Test as pallet_staking::Config>::SessionsPerEra::get() as u64;
+        let epoch_duration = SESSION_DURATION;
+        let era_duration = sessions_per_era * epoch_duration;
+
+        let (initial_total_issuance, _, _, initial_rewards_pool_balance) = chain_state();
+        assert_eq!(initial_rewards_pool_balance, 0); // ED is auto-deducted by the getter function
+
+        // Running chain until era rollover
+        run_to_block(era_duration + 1);
```

Comment:
you can just skip blocks by setting new bn
please do the same for already existing tests

Reply:
It won't work: we need authorship and session information to be updated block by block. Not in this particular test maybe but in other tests in this pallet. I'd keep it this way for the sake of uniformity.

Anyway, removed irrelevant hooks invocations.